File: 241027/그래프 탐색/graph-traversal.java
import java.util.*;

public class Main {
    public static final int MAX_N = 1001;
    public static int cnt;
    public static boolean[] visited = new boolean[MAX_N];
    public static ArrayList<Integer>[] graph = new ArrayList[MAX_N];
    
    public static void dfs(int vertex) {
        for (int i = 0; i < graph[vertex].size(); i++) {
            int current = graph[vertex].get(i);
            if (!visited[current]) {
                visited[current] = true;
                cnt += 1;
                dfs(current);
            }
        }
    }

    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        for (int i = 0; i < MAX_N; i++) {
            graph[i] = new ArrayList<>();
        }

        while (m-- > 0) {
            int x = sc.nextInt();
            int y = sc.nextInt();

            graph[x].add(y);
            graph[y].add(x); 
        }

        visited[1] = true;
        dfs(1);

        System.out.println(cnt);
    }
}

File: 240830/그래프 탐색/graph-traversal.py
n, m = map(int, input().split())

graph = [[] for _ in range(n + 1)]
visited = [False for _ in range(n + 1)]
for _ in range(m):
    x, y = map(int, input().split())
    graph[x].append(y)
    graph[y].append(x)

cnt = 0
def dfs(vertex):
    global cnt

    for curr_v in graph[vertex]:
        if not visited[curr_v]:
            cnt += 1
            visited[curr_v] = True
            dfs(curr_v)

visited[1] = True
dfs(1)

print(cnt)